.github/workflows/build.yml

```diff
@@ -12,23 +12,58 @@ jobs:
     steps:
       - name: Cancel previous running workflows
         uses: fkirc/skip-duplicate-actions@9da67cec51d21092667038f0f4fda73099a3a451
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
-      - uses: actions/setup-node@v2
+      - uses: actions/setup-node@v3
         with:
           node-version: 14.17.6
-      - uses: actions/cache@v2
+          cache: 'yarn'
+
+      - name: Cache node_modules
+        uses: actions/cache@v3
+        id: cache-primes
         with:
           path: "**/node_modules"
           key: ${{ runner.os }}-node_modules-${{ hashFiles('**/yarn.lock') }}
-      - name: install dependencies
+
+      - name: Install dependencies
+        if: steps.cache-primes.outputs.cache-hit != 'true'
         run: yarn install --frozen-lockfile --non-interactive --production=false
+
       - name: Lint commit message
         uses: wagoid/commitlint-github-action@296247dfa6a133767fe0d64d08fb66326047b680
-      - name: lint Javascript
+
+      - name: Lint javascript and typescript
         run: yarn lint
 
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    needs: [lint]
+    steps:
+      - name: Cancel previous running workflows
+        uses: fkirc/skip-duplicate-actions@master
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14.17.6
+          cache: 'yarn'
+
+      - name: Cache node_modules
+        uses: actions/cache@v3
+        id: cache-primes
```

Comment:
Some actions use this ID key, and some other does not, is it intended ?

Reply:
I think I got it, this is an identifier to get the step and assert cache has been found. Correct ? If cache has been found based on yarn.lock hash, get it, otherwise install dependencies. It is a speed up measure.

Reply:
That being said, knowing that any action requires the linter, I would say that force reinstalling the dependencies only in the lint step should be enough. WDYT ?

Reply:
This is used to check we we fetched a cache or not, which allows to decide if we run `yarn install`.
The steps that use it are required for the "test" and "deploy" steps.
Since those two require either lint or build to start, we're sure to have a cache when they are triggered, so we don't execute `yarn install` and we don't need the key to check if we properly fetched a cache.

Reply:
Ok following up my next message: 

> That being said, knowing that any action requires the linter, I would say that force reinstalling the dependencies only in the lint step should be enough. WDYT ?

This means that only the action at the root of anything needs to reinstall dependencies if not cache are present. So we can remove the `yarn install` from the `build` action correct ? (every actions are based on the `lint` one)

Reply:
👍